android-exoplayer/src/main/java/com/brentvatne/exoplayer/ReactExoplayerView.java

```diff
@@ -1257,72 +1645,52 @@ public void onPlaybackParametersChanged(PlaybackParameters params) {
-                                decoderInitializationException.mimeType);
-                    } else {
-                        errorString = getResources().getString(R.string.error_no_decoder,
-                                decoderInitializationException.mimeType);
-                    }
-                } else {
-                    errorString = getResources().getString(R.string.error_instantiating_decoder,
-                            decoderInitializationException.codecInfo.name);
-                }
-            }
+    public void onIsPlayingChanged(boolean isPlaying) {
+        eventEmitter.playbackStateChanged(isPlaying);
+    }
+
+    @Override
+    public void onPlayerError(PlaybackException e) {
+        if (e == null) {
+            return;
         }
-        else if (e.type == ExoPlaybackException.TYPE_SOURCE) {
-            errorString = getResources().getString(R.string.unrecognized_media_format);
+        String errorString = "ExoPlaybackException: " + PlaybackException.getErrorCodeName(e.errorCode);
+        String errorCode = "2" + String.valueOf(e.errorCode);
+        boolean needsReInitialization = false;
+        switch(e.errorCode) {
+            case PlaybackException.ERROR_CODE_DRM_DEVICE_REVOKED:
+            case PlaybackException.ERROR_CODE_DRM_LICENSE_ACQUISITION_FAILED:
+            case PlaybackException.ERROR_CODE_DRM_PROVISIONING_FAILED:
+            case PlaybackException.ERROR_CODE_DRM_SYSTEM_ERROR:
+            case PlaybackException.ERROR_CODE_DRM_UNSPECIFIED:
+                if (!hasDrmFailed) {
+                    // When DRM fails to reach the app level certificate server it will fail with a source error so we assume that it is DRM related and try one more time
+                    hasDrmFailed = true;
+                    playerNeedsSource = true;
+                    updateResumePosition();
+                    initializePlayer();
+                    setPlayWhenReady(true);
+                    return;
+                }
+                break;
+            default:
+                break;
         }
-        eventEmitter.error(errorString, ex);
+        eventEmitter.error(errorString, e, errorCode);
         playerNeedsSource = true;
-
-//        if (isBehindLiveWindow(e)) {
-//            // retry after 5 seconds
-//            new Handler().postDelayed(
-//                    new Runnable() {
-//                        public void run() {
-//                            if (player != null) {
-//                                currentlyInRetry = true; // to disable to youbora
-//                                clearResumePosition();
-//                                initializePlayer();
-//                            }
-//                        }
-//                    }, 5000);
-//        } else {
-//            updateResumePosition();
-//        }
-
-        if (!isBehindLiveWindow(e)) {
+        if (isBehindLiveWindow(e)) {
+            clearResumePosition();
+            initializePlayer();
```

Comment:
put an extra condition here and add a variable to check if we already did retry after isBehindLiveWindow has happened similar to `hasDrmFailed` .

call it `didBehindLiveWindowHappen`
so it doesnt keep retrying by calling `initializePlayer`